pkg/block/gs/adapter.go

```diff
@@ -26,10 +25,7 @@ const (
 )
 
 var (
-	ErrMismatchPartETag    = errors.New("mismatch part ETag")
-	ErrMismatchPartName    = errors.New("mismatch part name")
```

Comment:
Do we need to still return this for compatability?

Reply:
So we are not 100% S3 compatible with regard to errors. See documentation here: https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html
https://docs.aws.amazon.com/AmazonS3/latest/API/API_CompleteMultipartUpload.html

For example, if the ETAG doesn't match, there is a specific error:
Error Code: InvalidPart
Description: One or more of the specified parts could not be found. The part might not have been uploaded, or the specified ETag might not have matched the uploaded part's ETag.
HTTP Status Code: 400 Bad Request

Our code contains this error but does not invoke it, see pkg/gateway/errors/errors.go
There is an ErrInvalidPart but no one uses it